### ISSUE TITLE
Feature request:  Use response files for rustc build/doc.

### ISSUE THREAD

**Issue by MaulingMonkey** [rust-lang/cargo#7759]
(**EDIT:** rustc response files are not yet stabilized, but it looks like they will be in [1.41](https://github.com/rust-lang/rust/pull/66172)?)

**Describe the problem you are trying to solve**

Some crates have [hundreds](https://docs.rs/crate/winapi/0.3.8/source/Cargo.toml.orig) upon [hundreds](https://docs.rs/crate/web-sys/0.3.33/source/Cargo.toml.orig) - if not [thousands](https://docs.rs/crate/jni-android-sys/0.0.9/source/Cargo.toml.orig) - of features to manage compile times.  If writing wildcard features that include a significant subset of these features, it's easy to exceed command line length limits on Windows:

```
error: could not compile `jni-android-sys`.

Caused by:
  could not execute process `rustc --edition=2018 --crate-name ...
  ...thousands of --cfg "feature=\"x-y-z\"" flags...
  ... (never executed)

Caused by:
  The filename or extension is too long. (os error 206)
```

As an awkward workaround, you can have a feature simply not specify the features it depends on, and instead do them all in code.  E.g. instead of:
```rust
#[cfg(feature = "java-lang-Object")]
```
I've been generating:
```rust
#[cfg(any(feature = "all", feature = "java", feature = "java-lang", feature = "java-lang-Object"))]
```
But this has to be done *every* time you want to check the existence of java::lang::Object, and this isn't even the worst version.

**Describe the solution you'd like**

`rustc` already supports [@path](https://doc.rust-lang.org/rustc/command-line-arguments.html#path-load-command-line-flags-from-a-path) arguments / response files... so we could just leverage those.  The main annoyances are refactoring and figuring out what commands need to use response files ("all rustc invokes"?)

**Notes**

I'm [taking a quick stab at](https://github.com/rust-lang/cargo/compare/master...MaulingMonkey:pr-response-files) seeing how easy this is to implement... by which I mean all the unit tests are failing and it's probably my fault...

**Comment by epage** [comment]
This sounds like a duplicate of #1241 which I think was resolved in #10546.

@weihanglo does it look like we can close this?

**Comment by weihanglo** [comment]
I believe it was resolved in #10546. Closing.
